src/oops/designPatterns/prototype/Registry.java

```diff
@@ -49,4 +51,13 @@ private void loadItems() {
 		book.setTitle("Basic Book");
 		items.put("Book", book);
 	}
+
+	public static Registry getInstance() {
+		// TODO Auto-generated method stub
+		if(registeryInstance==null)
+		{
+			registeryInstance = new Registry();
```

Comment:
In this case, this block of code will never get executed as you have already stored the object of Registry in registryInstance when jvm loaded your class(Class Loader). If you make the registryInstance as null then only it will get executed. I this block of code is correct but above you will have set registryInstace=null.